### ISSUE TITLE
Issue with Server Maybe?

### ISSUE THREAD

**Issue by nebhead** [weberbox/PiFire-Android#4]
### Describe the bug

OK, I am not sure if this is expected behavior, but when I was trying to setup the Android App today - I found that I couldn't manually enable OneSignal in the WebUI settings page.  

![image](https://user-images.githubusercontent.com/19876886/163279006-4a7fd3f1-6b66-4fca-bd89-a1a075d27858.png)

And when I went to look at the app.py code, I found that it only had the following for OneSignal setup - but no input for enabling. 

```python
		if('delete_device' in response):
			DeviceID = response['delete_device']
			settings['onesignal']['devices'].pop(DeviceID)

		if('edit_device' in response):
			if(response['edit_device'] != ''):
				DeviceID = response['edit_device']
				settings['onesignal']['devices'][DeviceID] = {
					'friendly_name' : response['FriendlyName_' + DeviceID],
					'device_name' : response['DeviceName_' + DeviceID],
					'app_version' : response['AppVersion_' + DeviceID]
				}
```

Or, was this by design?  I can get the app to connect to my server if I go and manually update my settings.  

### Steps to Reproduce

1. In PiFire WebUI, go to settings > notify
2. Toggle enable for OneSignal
3. Save

Try to connect via App, and it will fail.  

### PiFire Android Version

v1.4.1

### PiFire Server Version

v1.3.1

### Mobile Device

Nexus 7

**Comment by weberbox** [comment]
Yes this is a bug. It is missing the onesignal toggle. I will need to update and request a PR . Hopefully things like this will be less likely to happen since I can now work off the same branch and not need to copy over from a different one. 

```
if('onesignal_enabled' in response):
			if(response['onesignal_enabled'] == 'on'):
				settings['onesignal']['enabled'] = True
		else:
			settings['onesignal']['enabled'] = False
```

As far as the app connecting that should work regardless of the onesignal setting in the server being on or off.  You can toggle onesignal off and on via the app once connected as that will send raw json to enable/disable it.  

If the app is not connecting it is something else. Is there any error when trying to connect? It should show why it cannot connect during the setup. 

I just connected the app to my fresh install on my zero w sitting at my desk without any issue and toggled onesignal off and on from the app.

**Comment by nebhead** [comment]
Ok, let me try again.  Maybe I'm doing the sequence wrong.  I had a red bar at the top of the screen when I first attempted to install, however I managed to enable OneSignal by editing the settings.json directly.  Then it seemed the app was connecting, but now it seems to not be fetching data.  No red bar this time.  Anyway, I'll try again and see if I can get some screen captures.

**Comment by weberbox** [comment]
Ok let me know what you find so we can get it up and working. The app does work without OneSignal as that is only for push notifications. If you download the non firebase version is does not even have the onesignal code at all. 

The rest of the functions are all just done through json communication over websockets. The app supports https with both self signed and ca signed. If the server uses self signed then the cert has to be imported into the Android key store or the app will not trust it.

Http should work with an IP or a dns address.

**Comment by nebhead** [comment]
Just tried the Firebase version of the app (was using the non-firebase version previously), and set it up using my old Nexus 6P, and it seems to be working fine now.  The Nexus 7 tablet which I was initially using, actually hung when trying to use the QR code then shut down.  :(  I think that's just due to the fact that it's super old.  Anyway, it seems to be working again.  Thanks for the help!

**Comment by weberbox** [comment]
Ok great!! I will check into the QR code scanning. I only have one Android tablet and it seemed to be working ok but sometimes cameras on tablets can act strange especially old ones.

**Comment by nebhead** [comment]
Yep, and not to mention, I have lineage OS running on that tablet, so it's a mixed bag of what hardware works and what doesn't.  The WiFi seems flaky too, which is likely the cause of the disconnects.  Anyway, the Nexus 6P seems to be just fine, so I'll chalk it up to old hardware with sketchy OS.

**Comment by nebhead** [comment]
Oh, you know, I'm using the non-firebase version right now.  I'll try the
firebase version and see if that makes a difference.  It also *could *be my
very, very old Nexus 7 tablet with Lineage OS.  It's had WiFi issues in the
past.  Might need to try it with a different device.
